pkg/parse/state.go

```diff
@@ -192,3 +202,51 @@ func (s *reconcilerState) needToSetSyncStatus(newStatus syncStatus) bool {
 	// Update if there's a diff
 	return !newStatus.equal(s.syncStatus)
 }
+
+// ResetRetryBackoffAndTimer resets the retry backoff and timer.
+func (s *ReconcilerState) ResetRetryBackoffAndTimer() {
+	s.retryBackoff = defaultBackoff()
+	s.retryTimer.Reset(s.retryPeriod)
+}
+
+// InitRetryTimerAndPeriod initializes the retry timer and period.
+func (s *ReconcilerState) InitRetryTimerAndPeriod(timer *time.Timer, period time.Duration) {
+	s.retryTimer = timer
+	s.retryPeriod = period
+}
+
+// NextRetryDuration returns the amount of time to sleep for next retry.
+// It also mutates the retryBackOff to update its Steps and Duration.
+func (s *ReconcilerState) NextRetryDuration() time.Duration {
+	return s.retryBackoff.Step()
+}
+
+// ReachRetryLimit returns whether the retry reaches the limit.
+func (s *ReconcilerState) ReachRetryLimit() bool {
+	return s.retryBackoff.Steps == 0
+}
+
+// RetrySteps returns the remaining number of retry iterations.
+func (s *ReconcilerState) RetrySteps() int {
+	return s.retryBackoff.Steps
+}
+
+// NeedRetry returns whether a retry is needed.
+func (s *ReconcilerState) NeedRetry() bool {
+	return s.cache.needToRetry
+}
+
+// SourceCommitInStatus returns the source commit in the status field.
+func (s *ReconcilerState) SourceCommitInStatus() string {
```

Comment:
suggest: 
- StatusSourceCommit
- StatusRenderingCommit
- StatusSyncCommit

Then it's closer to how it looks like the RSync api